### DISCUSSION TITLE
Truffle React Box - "window.ethereum" Undefined

### DISCUSSION THREAD

**Discussion by lloy0076** [ConsenSys-archive/truffle#6001] [Truffle Boxes]
I raised this issue:

- https://github.com/truffle-box/react-box/issues/199

In short, I unboxed the React box, started ganache, ran the migration (and the tests succeed too).

Then when I run the client (`npm run start`) and visit the URL in a browser, there is an error in any browser _without_ a MetaMask extension installed; but once I install it, the React app runs as expected.

It is possible one does need some sort of Wallet extension (which I think is what MetaMask is) for the code to work, but unfortunately it just "white screens" and there's no obvious documentation saying that such a thing is required.

There's more details in the ticket.

Is this just me?

**Comment by lloy0076** [comment]
Reading a bit further, MetaMask itself says you can detect the MetaMask by doing this:


```
  if (typeof window.ethereum !== 'undefined') {
    console.log('MetaMask is installed!');
  }
```

(see https://docs.metamask.io/wallet/get-started/detect-metamask)

And you could query `window.ethereum.isMetaMask` to see if it is a MetaMaskProvider.

I am not new to programming but am new to working with blockchains, solidity and so forth; so I _deduce_ that one might be able to have any type of provider *but* it is an error to have no provider.

**Comment by FlaviuWeb3** [comment]
Hi @lloy0076 our engineering team will look into this. Thanks for passing the above feedback with us.
